R/dash.R

```diff
@@ -702,7 +702,43 @@ Dash <- R6::R6Class(
       }
       private$callback_context_
     },
-
+    
+    # ------------------------------------------------------------------------
+    # return asset URLs
+    # ------------------------------------------------------------------------
+    get_asset_url = function(asset_path, prefix = "/") {
+      asset <- lapply(private$asset_map, 
+                      function(x) {
+                        # asset_path should be prepended with the full app root & assets path
+                        # if leading slash(es) present in asset_path, remove them before
+                        # assembling full asset path
+                        asset_path <- file.path(dirname(private$app_root_path),
```

Comment:
I encountered the following error when running this line to generate the `asset_path`:

`Error in dirname(private$app_root_path) : a character vector argument expected`

Digging a bit deeper into it, the `app_root_path` private field in Dash calls  the function `getAppPath` from `utils.R` upon initialization of the Dash class, and returns the value `FALSE` if the app isn't run via `source()`. 

A potential fix could be:
`ifelse(is.logical(private$app_root_path), "", dirname(private$app_root_path))`, but I'm not exactly sure what needs to be there when the condition is TRUE to get the correct path. 

A more permanent fix would be to tweak the `getAppPath` function to deliver a more comprehensive return value for the `else` condition.